starfish/test/dataset_fixtures.py

```diff
@@ -50,58 +50,51 @@ def default_tile_extras_provider(hyb: int, ch: int, z: int) -> Any:
 DEFAULT_WIDTH = 20
 
 
-@pytest.fixture(scope='session')
-def synthetic_stack_factory():
-    """
-    Inject this factory, which is a method to produce image stacks.
+def synthetic_stack(
```

Comment:
Now that it's a function, can we make these global-scope-inherited `Indices` into parameters with default values so that this can be called elsewhere?

Also please take a look at my synthetic data PR sometime soon -- using that as a `tile_data_provider` is probably the way we want to go for more robust testing of downstream components that need things like spots to be evaluated.

Reply:
> Now that it's a function, how would you feel about making these global-scope-inherited Indices into parameters with default values so that this can be called elsewhere?

Not sure I follow here.  Can you explain?

Reply:
Sorry, I was suggesting that synthetic stack could take as parameters `NUM_HYB, NUM_CH, ...`. Do you think that would be useful?

Reply:
it does...?

Reply:
https://github.com/spacetx/starfish/blob/f0d545a631cb95640729faa9e2765f24ee1921e4/starfish/test/dataset_fixtures.py#L59